v0/sources/shopify/transform.js

```diff
@@ -27,16 +37,37 @@ const ecomPayloadBuilder = (event, shopifyTopic) => {
   message.setEventType(EventType.TRACK);
   message.setEventName(RUDDER_ECOM_MAP[shopifyTopic]);
 
-  const properties = createPropertiesForEcomEvent(event);
+  let properties = createPropertiesForEcomEvent(event);
+  properties = removeUndefinedAndNullValues(properties);
   Object.keys(properties).forEach(key =>
     message.setProperty(`properties.${key}`, properties[key])
   );
+  // Map Customer details if present
+  const customerDetails = get(event, "customer");
+  if (customerDetails) {
+    message.setPropertiesV2(
+      customerDetails,
+      MAPPING_CATEGORIES[EventType.IDENTIFY]
+    );
+  }
   if (event.updated_at) {
     // TODO: look for created_at for checkout_create?
     // converting shopify updated_at timestamp to rudder timestamp format
     message.setTimestamp(new Date(event.updated_at).toISOString());
   }
-  if (event.user_id) {
+  if (event.customer) {
+    message.setPropertiesV2(
+      event.customer,
+      MAPPING_CATEGORIES[EventType.IDENTIFY]
+    );
+  }
+  if (event.shipping_address) {
+    message.setProperty("traits.shippingAddress", event.shipping_address);
+  }
+  if (event.billing_address) {
+    message.setProperty("traits.billingAddress", event.billing_address);
+  }
```

Comment:
Can't we make this using a switch case?

Reply:
Then we have to iterate every event properties, as both of these can be triggered independent of each other made if else logic here